base/show.jl

```diff
@@ -436,7 +436,8 @@ Julia code when possible.
 
 For a more verbose human-readable text output for objects of type `T`, define
 `show(io::IO, ::MIME"text/plain", ::T)` in addition. Checking the `:compact`
-[`IOContext`](@ref) property of `io` in such methods is recommended,
+[`IOContext`](@ref) key (often checked as `get(io, :compact, false)`)
```

Comment:
How about making it type stable?
```suggestion
[`IOContext`](@ref) key (often checked as `get(io, :compact, false)::Bool`)
```

Reply:
What do you think about https://github.com/JuliaLang/julia/pull/46828?